packages/amplify-cli/src/extensions/amplify-helpers/invoke-plugin-method.ts

```diff
@@ -0,0 +1,35 @@
+import { $TSContext } from 'amplify-cli-core';
+
+export const invokePluginMethod = async <T>(
+  context: $TSContext,
```

Comment:
I would follow KISS and yes, pass it in, the reason for "lot of context" is that we hang it on context, we don't manage `this` binding to anything (where we could use that) and this method is "independent" of context but this is where it is hanging in the current code. I would not change it and would not store context anywhere, as it can be an easy miss when we refactor it.